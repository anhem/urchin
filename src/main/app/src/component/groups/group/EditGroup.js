import React, {Component} from 'react'
import FieldError from "../../FieldError";
import history from "../../../history";

class Groups extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setGroup(data);
    };

    del = () => {
        this.props.callbacks.deleteGroup(this.props.groupId);
    };

    back = () => {
        history.push('/groups');
    };

    render() {
        let groupName = this.props.group.groupName || '';
        return (

            <div>
                <h2>Group</h2>
                <input
                    name="groupName"
                    type="text"
                    value={groupName}
                    onChange={this.update}
                />
                <FieldError
                    fieldErrors={this.props.fieldErrors}
                    field="groupName"
                />
                <button onClick={this.del}>Delete</button>
                <button onClick={this.back}>Back</button>
            </div>
        )
    }
}

export default Groups