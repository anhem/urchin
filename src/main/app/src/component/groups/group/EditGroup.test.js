import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import EditGroup from "./EditGroup";

describe('EditGroup', () => {

    let props = {
        group: {},
        usersInGroup: [],
        availableUsers: []
    };

    it('match snapshot', () => {
        expect(toJson(shallow(<EditGroup {...props}/>))).toMatchSnapshot();
    });
});
