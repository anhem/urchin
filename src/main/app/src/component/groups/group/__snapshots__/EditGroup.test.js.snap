// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGroup match snapshot 1`] = `
<div
  className="edit-group"
  data-view="editGroup"
>
  <h2>
    Group
  </h2>
  <div>
    name: 
  </div>
  <select
    className="edit-group__available-users"
    data-view="availableUsers"
    name="userId"
    onChange={[Function]}
  >
    <option>
      -- select user --
    </option>
  </select>
  <button
    className="edit-group__add-user-btn"
    data-view="addUser"
    onClick={[Function]}
  >
    Add User
  </button>
  <button
    className="edit-group__delete-btn"
    data-view="delete"
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    className="edit-group__back-btn"
    data-view="back"
    onClick={[Function]}
  >
    Back
  </button>
  <h2>
    Members of group
  </h2>
  <ul
    className="edit-group__users"
  />
</div>
`;
